rpc/test/helpers.go

```diff
@@ -170,17 +179,71 @@ func NewTendermint(app abci.Application, opts *Options) *nm.Node {
 	if err != nil {
 		panic(err)
 	}
+
+	config.IPFS = cfg.DefaultIPFSConfig()
+	err = initIpfs(config, opts.loadIpfsPlugins, logger)
+	if err != nil {
+		panic(err)
+	}
 	node, err := nm.NewNode(config, pv, nodeKey, papp,
 		nm.DefaultGenesisDocProviderFunc(config),
 		nm.DefaultDBProvider,
 		nm.DefaultMetricsProvider(config.Instrumentation),
-		logger)
+		logger,
+		nm.IpfsPluginsWereLoaded(true),
+	)
 	if err != nil {
 		panic(err)
 	}
 	return node
 }
 
+func initIpfs(config *cfg.Config, loadPlugins bool, log log.Logger) error { // add counter part in ResetAllCmd
+	// init IPFS config with params from config.IPFS
+	// and store in config.IPFS.ConfigRootPath
+	repoRoot := config.IPFSRepoRoot()
+	if fsrepo.IsInitialized(repoRoot) {
+		log.Info("ipfs repo already initialized", "repo-root", repoRoot)
+		return nil
+	}
+	var conf *ipfscfg.Config
+
+	identity, err := ipfscfg.CreateIdentity(ioutil.Discard, []options.KeyGenerateOption{
+		options.Key.Type(options.Ed25519Key),
+	})
+	if err != nil {
+		return err
+	}
+	if err := tmos.EnsureDir(repoRoot, 0700); err != nil {
+		return err
+	}
+	if loadPlugins {
+		plugins, err := loader.NewPluginLoader(filepath.Join(repoRoot, "plugins"))
+		if err != nil {
+			return fmt.Errorf("error loading plugins: %s", err)
+		}
+		if err := plugins.Load(&nodes.LazyLedgerPlugin{}); err != nil {
+			return err
+		}
+		if err := plugins.Initialize(); err != nil {
+			return fmt.Errorf("error initializing plugins: %s", err)
+		}
+		if err := plugins.Inject(); err != nil {
+			return fmt.Errorf("error initializing plugins: %s", err)
+		}
```

Comment:
This looks identical to `setupPlugins()` in `node/node.go`?

Reply:
It is indeed. I didn't want to have to import the node package here. As it is just tests, I don't mind the duplication.

Reply:
Me right now: ಠ_ಠ, but okay

Reply:
I'm more and more convinced that the ipfs repo intitialization shouldn't be part of `tendermint init` even if that looks right on first sight. Instead `tendermint start` should actually take the tendermint config and create, and initialize the repo (and config) if necessary (using some options from the tendermint config). That way, it wouldn't be necessary to have that method here, too.